synapse/rest/client/v1/login.py

```diff
@@ -96,6 +97,9 @@ def on_GET(self, request):
         flows = []
         if self.jwt_enabled:
             flows.append({"type": LoginRestServlet.JWT_TYPE})
+        if self.oidc_enabled:
+            flows.append({"type": LoginRestServlet.SSO_TYPE})
+            flows.append({"type": LoginRestServlet.TOKEN_TYPE})
         if self.saml2_enabled:
             flows.append({"type": LoginRestServlet.SSO_TYPE})
             flows.append({"type": LoginRestServlet.TOKEN_TYPE})
```

Comment:
I suspect that these should be if-elif statements as we use below in `register_servlets` (essentially meaning you can only have CAS, SAML, or OIDC enabled). This means we should order them the same as below (CAS, SAML, then ODIC).

Reply:
Would it not make sense to be able to support several simultaneously?

Reply:
I'm unsure if it would make sense, if it would I think that would be a follow-up. I think it is more important to ensure *something* works if you have a weird config.